### PR TITLE
Fix some CLI errors

### LINKED ISSUE #59: CLI errors
image

This seems to happen when no other arg is supplied to the program (so `sys.argv[1]` throws an error)
image

### PR DESCRIPTION
**Pull request by Chiroyce1**
Fixes #59 

- Adds `.gitignore` so it doesn't commit the `__pycache__` folder generated by Python when importing files/modules
- Adds a user-agent header to the request sent for scraping follower count (without UA, Scratch returns blank HTML)
- Fixes error that occurs when no argument is provided in CLI (so it instead shows the help message)

Also sorry my formatter kicked in and made a bunch of changes, you can revert them if you want.

### CODE REVIEW

File: scratchclient/__main__.py [hunk 1]
```diff
@@ -228,6 +228,9 @@ def forum_category_data(category_id):
             print("\n{0}\n".format(separator()))
 
 
+if len(sys.argv) == 1:
+    help()
+    exit()
```
Comment (CubeyTheCube) [inline scratchclient/__main__.py:233]: Add an empty line after this

**Review by CubeyTheCube — CHANGES_REQUESTED** [review]

File: scratchclient/Scraping.py [hunk 2]
```diff
@@ -403,25 +415,29 @@ def rgb_to_hex(rgb):
...
+        page = requests.get(
+            f"https://scratch.mit.edu/users/{username}/followers/", headers=self.headers).text
         return re.search("&raquo;\s+Followers\s+\((\d+)\)", page).groups()[0]
 
     def get_following_count(self, user):
         username = user if isinstance(user, str) else user.username
 
-        page = requests.get(f"https://scratch.mit.edu/users/{username}/following/").text
+        page = requests.get(
+            f"https://scratch.mit.edu/users/{username}/following/").text
```
[the remark this answers is not in the document]
Comment (Chiroyce1) [reply scratchclient/Scraping.py:432]: The other pages don't require it (as in they return proper content even without the header) but I'll still add it.

**Comment by Chiroyce1** [comment]
@CubeyTheCube done.

File: scratchclient/Scraping.py [hunk 3]
```diff
@@ -402,57 +402,60 @@ def rgb_to_hex(rgb):
 # For lack of a better name
 class ScrapingSession:
     def __init__(self, client):
+        self._headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36."}
```
Comment (CubeyTheCube) [inline scratchclient/Scraping.py:405]: There should be a new line and an indent after the brace

**Review by CubeyTheCube — CHANGES_REQUESTED** [review]

**Review by CubeyTheCube — APPROVED** [review]

File: scratchclient/Scraping.py [hunk 4]
```diff
@@ -402,57 +402,61 @@ def rgb_to_hex(rgb):
 # For lack of a better name
 class ScrapingSession:
     def __init__(self, client):
+        self._headers = {
```
Comment (CubeyTheCube) [inline scratchclient/Scraping.py:405]: Oops, misclicked the "accept" button. I meant for this to be 
```py
self._headers = {
  "User-Agent": "user agent here"
}
```
lmao

**Review by CubeyTheCube — CHANGES_REQUESTED** [review]

File: scratchclient/Scraping.py (hunk 4 above)
Comment (Chiroyce1) [reply scratchclient/Scraping.py:405]: lmao ok done - btw why formatting for this but not any other line? 👀

**Pull request merged**